modules/tinymce/src/core/main/ts/init/LicenseKeyManager.ts

```diff
@@ -0,0 +1,133 @@
+import { Obj, Type } from '@ephox/katamari';
+
+import AddOnManager, { AddOnConstructor } from '../api/AddOnManager';
+import Editor from '../api/Editor';
+import * as Options from '../api/Options';
+
+import * as ForceReadonly from './ForceReadonly';
+
+export type LicenseKeyManagerAddon = AddOnConstructor<LicenseKeyManager>;
+
+interface LicenseKeyManagerLoader {
+  readonly load: (editor: Editor, suffix: string) => void;
+  readonly isLoaded: (editor: Editor) => boolean;
+  readonly add: (addOn: LicenseKeyManagerAddon) => Promise<void>;
+  readonly init: (editor: Editor) => void;
+}
+
+interface ValidateData {
+  plugin?: string;
+  [key: string]: any;
+}
+
+// TODO: Can we do better than just a Promise<boolean>?
+export interface LicenseKeyManager {
+  readonly verify?: (editor: Editor) => Promise<boolean>;
+  readonly validate?: (editor: Editor, data?: ValidateData) => Promise<boolean>;
+}
+
+const displayNotification = (editor: Editor, message: string) => {
+  editor.notificationManager.open({
+    type: 'error',
+    text: message
+  });
+};
+
+const GplLicenseKeyManager: LicenseKeyManager = {
+  verify: () => Promise.resolve(true),
+  validate: (_editor, data) => {
+    const { plugin } = data || {};
+    // Premium plugins are not allowed if 'gpl' is given as the license_key
+    if (Type.isString(plugin)) {
+      return Promise.resolve(false);
+    } else {
+      return Promise.resolve(true);
+    }
```

Comment:
I'm not sure it makes sense to use this for anything other than plugins, but if you do want this it can simplify to
```suggestion
    const plugin = data?.plugin;
    // Premium plugins are not allowed if 'gpl' is given as the license_key
    return Promise.resolve(!Type.isString(plugin))
```

Reply:
Thanks, that is a bit more compact 👍